client/my-sites/promote-post-i2/components/main-wrapper/index.tsx

```diff
@@ -16,6 +16,17 @@ export default function MainWrapper( { children }: Props ) {
 	const sourceQuery = currentQuery?.[ 'source' ];
 	const source = sourceQuery ? sourceQuery.toString() : undefined;
 
+	const type = keyValue?.split( '-' )[ 0 ];
```

Comment:
We can include this logic in the `usePromoteParams` hook. It is already calculating the post ID from the keyValue, so changing that logic and adding the new campaign Id would be fine.